app/models/project.rb

```diff
@@ -47,6 +51,15 @@ def tasks
     ProjectConfiguration.templates[template]::TASKS
   end
 
+  def available_extensions
+    AVAILABLE_EXTENSIONS
+  end
+
+  def extensions
+    attributes["extensions"].reject! {|a| a == ''} unless attributes["extensions"].nil?
```

Comment:
`unless foo.nil?` is basically the same as `if foo`, and the latter should be preferred for readability.